submissions/pashapushak1996/oop-exercise/index.js

```diff
@@ -0,0 +1,55 @@
+class Inhabitant {
+    constructor(name, gender, greeting) {
+        this.species = this.constructor.name.toLowerCase();
+        this.name = name;
+        this.gender = gender;
+        this.greeting = greeting;
+    }
+}
+
+class Human extends Inhabitant {
+    constructor(name, gender, greeting) {
+        super(name, gender, greeting);
+        this.legs = 2;
+        this.hands = 2;
+    }
+}
+
+class Animal extends Inhabitant {
+    constructor(name, gender, greeting) {
+        super(name, gender, greeting);
+        this.paws = 4;
+    }
+}
+
+class Cat extends Animal {
+    constructor(name, gender, greeting) {
+        super(name, gender, greeting);
+    }
+}
+
+class Dog extends Animal {
+    constructor(name, gender, greeting) {
+        super(name, gender, greeting);
+    }
+}
+
+const dog = new Dog('Tom', 'male', 'Howl!');
+const cat = new Cat('Kitty', 'female', 'Meow!');
+const woman = new Human('Jessy', 'female', 'Hello!');
+const man = new Human('Ted', 'male', 'Hello my dear!');
+
+const inhabitants = [dog, cat, woman, man];
+
+const inhabitantsKeys = ['species', 'name', 'gender', 'greeting', 'legs', 'hands', 'paws'];
+
+const inhabitantValuesToString = (inhabitant) => {
+    return inhabitantsKeys
+        .filter((key) => inhabitant[key])
```

Comment:
`key` is a generic term. Living beings are characterized by properties or features.